server/controllers/helper.js

```diff
@@ -0,0 +1,21 @@
+const User = require("../models/User");
+
+exports.checkUserExists = async (email, username, res) => {
+  if (email) {
+    const emailExists = await User.findOne({ email });
+
+    if (emailExists) {
+      res.status(400);
+      throw new Error("A user with that email already exists");
+    }
+  }
+
+  if (username) {
+    const usernameExists = await User.findOne({ username });
+
+    if (usernameExists) {
+      res.status(400);
+      throw new Error("A user with that username already exists");
+    }
+  }
+};
```

Comment:
Similar comments about single-responsibility here. Ideally this function is only concerned about finding out whether the user exists or not - setting status codes and/or sending responses shouldn't be its responsibility.

Reply:
will try to create a workaround that you might approve.